modules/@apostrophecms/schema/index.js

```diff
@@ -612,90 +612,81 @@ module.exports = {
 
           const { convert } = self.fieldTypes[field.type];
 
-          if (convert) {
-            try {
-              const isAllParentsVisible = isParentVisible === false
-                ? false
-                : await self.isVisible(req, schema, destination, field.name);
-              const isRequired = await self.isFieldRequired(req, field, destination);
-              await convert(
-                req,
-                {
-                  ...field,
-                  required: isRequired
-                },
-                data,
-                destination,
-                {
-                  ...options,
-                  isParentVisible: isAllParentsVisible
-                }
-              );
-            } catch (error) {
-              if (Array.isArray(error)) {
-                const invalid = self.apos.error('invalid', {
-                  errors: error
-                });
-                invalid.path = field.name;
-                errors.push(invalid);
-              } else {
-                error.path = field.name;
-                errors.push(error);
-              }
-            }
+          if (!convert) {
+            continue;
           }
-        }
 
-        const errorsList = [];
+          const isCurrentVisible = isParentVisible === false
+            ? false
+            : await self.isVisible(req, schema, data, field.name);
```

Comment:
Notable difference here (previous versions might be buggy). We pass data instead of destination because destination is not fully built at this point.
I think `isVisible` should compare data as they have been set by  the user and not the potential default values.
Tests are green with it but you might have some reasons to not allow this.
Let me know.

Reply:
Data can be unvalidated stuff of unexpected types etc., it could be anything. Trusting it is problematic. You might get away with that in `isVisible` because you're not saving the values, but then `isVisible` must be written carefully to not assume the values are at all well-formed, for instance it must not assume string fields are actually strings yet, etc. because nothing has been validated.

Reply:
It now uses destination when eveything as been converted.